ssl/record/ssl3_record.c

```diff
@@ -490,6 +490,15 @@ int ssl3_get_record(SSL *s)
             goto f_err;
         }
 
+#ifndef OPENSSL_NO_TLSEXT
+        if (s->session && USE_MAX_FRAGMENT_LENGTH_EXT(s->session)
+                && rr[j].length > GET_MAX_FRAGMENT_LENGTH(s->session)) {
```

Comment:
RFC 6066 says that we MUST check the sizes of incoming records, adjust them for overheads such as padding etc, and them discard them with a record overflow alert before decryption.

Also, I don't see you limiting this extension to TLS only, but I don't see an equivalent check in DTLS.